once_test.py

```diff
@@ -402,32 +533,30 @@ def value(self):  # pylint: disable=inconsistent-return-statements
     def test_once_per_instance_do_not_block_each_other(self):
         class _BlockableClass:
             def __init__(self, test: unittest.TestCase):
-                self.lock = threading.Lock()
                 self.test = test
                 self.started = False
                 self.counter = Counter()
 
             @once.once_per_instance
             def run(self) -> int:
                 self.started = True
-                with self.lock:
-                    pass
                 return self.counter.get_incremented()
 
         a = _BlockableClass(self)
+        a.counter.paused = True
         b = _BlockableClass(self)
         with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
-            with a.lock:
-                a_job = executor.submit(a.run)
-                while not a.started:
-                    pass
-                # At this point, the A job has started. However, it cannot
-                # complete while we hold its lock. Despite this, we want to ensure
-                # that B can still run.
-                b_job = executor.submit(b.run)
-                # The b_job will deadlock and this will fail if different
-                # object executions block each other.
-                self.assertEqual(b_job.result(timeout=5), 1)
+            a_job = executor.submit(a.run)
```

Comment:
You can turn a.started into a threading.Event and another threading.Event for a.can_finish